internal/repository/prisma/step_run.go

```diff
@@ -452,22 +452,29 @@ func (s *stepRunRepository) QueueStepRun(ctx context.Context, tenantId, stepRunI
 			return repository.ErrStepRunIsNotPending
 		}
 
-		stepRun, err = s.updateStepRunCore(ctx, tx, tenantId, updateParams, updateJobRunLookupDataParams)
-
+		sr, err := s.updateStepRunCore(ctx, tx, tenantId, updateParams, updateJobRunLookupDataParams)
 		if err != nil {
 			return err
 		}
 
+		stepRun = sr
+
 		if err != nil {
 			return err
 		}
 
-		err = tx.Commit(context.Background())
+		if err := tx.Commit(context.Background()); err != nil {
+			return err
+		}
 
-		return err
+		return nil
 	})
 
-	err = retrier(s.l, func() error {
+	if retrierErr != nil {
+		return nil, fmt.Errorf("could not queue step run: %w", err)
+	}
```

Comment:
please double check @abelanger5 that this is intended.

Reply:
Yeah this looks right.